mpi4jax/_src/utils.py

```diff
@@ -176,12 +178,31 @@ def has_sycl_support() -> bool:
 
 def prefer_notoken() -> bool:
     """Returns True if primitive implementations should prefer not to use tokens."""
-    use_notoken = os.environ.get("MPI4JAX_PREFER_NOTOKEN", "1").lower() in (
-        "1",
-        "true",
-        "on",
-    )
-    if not use_notoken:
+
+    # No-token mode only works with JAX >= 0.5.1. See:
+    # - https://github.com/mpi4jax/mpi4jax/issues/274
+    # - https://github.com/jax-ml/jax/issues/26087
+    jax_ver_sufficient = versiontuple(jax.__version__) >= (0, 5, 1)
```

Comment:
We need to start earlier:

```suggestion
    jax_ver_sufficient = versiontuple(jax.__version__) >= (0, 5, 0)
```

`v0.5.0` doesn't work with tokens (see #273).

Reply:
(Let's not forget to update the references to 0.5.1 elsewhere too)

Reply:
No, it has to be `jax_ver_sufficient >= (0,5,1)` since 0.5.0 still has a bug.

> (Let's not forget to update the references to 0.5.1 elsewhere too)
I have found two places where we reference JAX version:
1. setup.py
We have `JAX_MINIMUM_VERSION = "0.4.35"` but if we set it to 0.5.1 it will not install it on Intel Macs, right?
One could issue this patch and once you drop the token mode completely also increase the minimum JAX version.

2. jax_compat.py
Function `check_jax_version()` only issues a warning if JAX version is too high.
Perhaps we could add a warning for users with JAX versions below 0.5.1, informing them that we will default to token mode and that in the future support for versions below 0.5.1 will be discontinued?
However, if the user is already on the latest available but outdated version, the warning is not helpful.

Reply:
I get that v0.5.0 is buggy when using with custom linear solves. However, **all versions past v0.5.0 do not work at all with notoken disabled**.  So `jax_ver_sufficient` must use `(0, 5, 0)`, and all other references to v0.5.1 _within this function_ should be updated.

Reply:
yeah, I agree with @dionhaefner . 
I think the minimum jax version should be 0.5.0. 
(@macekmar netket+mpi is broken in jax 0.5.0, there's nothing you can do. It's a netket problem, not a mpi4jax problem)

Reply:
If we have a test using custom linear solvers, we can simply skip it if jax==0.5.0